simple-web-app/app/src/main/java/by/academy/web/repos/PersonRepositoryInMemory.java

```diff
@@ -0,0 +1,199 @@
+package by.academy.web.repos;
+
+import by.academy.web.model.*;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Slf4j
+public class PersonRepositoryInMemory implements PersonRepository {
+
+    private final Map<Integer, Person> map = new ConcurrentHashMap<>();
+
+    private static volatile PersonRepositoryInMemory instance;
+
+    private PersonRepositoryInMemory() {
+        //singleton
+    }
+
+    public static PersonRepositoryInMemory getInstance() {
+        if (instance == null) {
+            synchronized (PersonRepositoryInMemory.class) {
+                if (instance == null) {
+                    instance = new PersonRepositoryInMemory();
+                }
+            }
+        }
+        return instance;
+    }
+
+    @Override
+    public List<Person> findAll() {
+        if (map.isEmpty()) {
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(map.values());
+    }
+
+    @Override
+    public Optional<Person> find(int id) {
+        return Optional.ofNullable(map.get(id));
+    }
+
+
+
+    @Override
+    public Person save(Person person) {
+        Integer id = person.getId();
+        if(person.getId()==1){
```

Comment:
Во первых, это повтор, ты же уже вытащил id строку назад. Используй его.
Во вторых, нельзя сравнивать Integer с 1. Это потенциальный NullPointerException, и он, кстати выпадает, если попробовать добавить кого-нибудь, например у меня в При добавлении на странице Teacher падает NPE.
В третьих, между операторами и операндами должны быть пробелы. Таже пробел должен быть после if и перед закрывающей скобкой. Пробегись по коду и поправь такие моменты.
```suggestion
        if (id != null && id == 1){
```

Reply:
убрал, кривая реализация моя.